### PR TITLE
Firebase in Datastore mode databases and locations

### PR DESCRIPTION
**Pull request by varunalla**
Databse and location permissions

### CODE REVIEW

File: services/gcp/datastore/locations.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Locations
+description: >-
+  Physical Locations of the Datastore instances.
+scope: LOW
+privileges:
+  get:
+    vulnerabilities: []
```
[the remark this answers is not in the document]
Comment (nbrahms) [reply services/gcp/datastore/locations.yml:7]: Mind adding a note?

File: services/gcp/datastore/databases.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Databases
+description: >-
+  Databases are the top level containers for data in Datastore mode. A datastore mode database is a type of database 
+  that uses the Google Cloud Datastore API to access data stored in Firestore databases.
+scope: MEDIUM
+notes: >-
+  Datastore mode quotas can be viewed in the GCP console at https://console.cloud.google.com/appengine/quotadetails.
+privileges:
+  get:
+    vulnerabilities: []
```
[the remark this answers is not in the document]
Comment (varunalla) [reply services/gcp/datastore/databases.yml:10]: get permission provides you a way to start and rollback transactions only.

File: services/gcp/datastore/databases.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    notes: >-
+      Principal with this permission can export entities to a Cloud Storage bucket.
+  import:
+    vulnerabilities: [impact:consumption, impact:spend, destruction:data]
+    notes: >-
+      Principal with this permission can import entities from a file, this can consume the quota limits for entity writes and storage.
+      Existing entities with the same keys are overwritten resulting in destruction of data.
+  create:
+    vulnerabilities: []
+  update:
```
[the remark this answers is not in the document]
Comment (varunalla) [reply services/gcp/datastore/databases.yml:34]: update the database configuration. added more information with reference to https://cloud.google.com/datastore/docs/manage-databases

File: services/gcp/datastore/databases.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  Databases are the top level containers for data in Datastore mode. A datastore mode database is a type of database 
+  that uses the Google Cloud Datastore API to access data stored in Firestore databases.
+scope: MEDIUM
+notes: >-
+  Datastore mode quotas can be viewed in the GCP console at https://console.cloud.google.com/appengine/quotadetails.
+privileges:
+  get:
+    vulnerabilities: []
+    notes: >-
+      Principal with this permission can begin and rollback transaction. the privilege by itself has no impact, given transactions are isolated.
```
Comment (nbrahms) [inline services/gcp/datastore/databases.yml:12]: Grammar:
```suggestion
      Principal with this permission can begin and rollback transaction. The privilege by itself has no impact, given transactions are isolated.
```

File: services/gcp/datastore/databases.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  getMetadata:
+    vulnerabilities: [discovery:metadata]
+    notes: >-
+      Principal with this permission can get metadata of the database.
```
Comment (nbrahms) [inline services/gcp/datastore/databases.yml:16]: Grammar:
```suggestion
      A principal with this permission can read the metadata of the database.
```

Also, is there anything sensitive in these metadat?

File: services/gcp/datastore/databases.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  list:
+    vulnerabilities: [discovery:infra]
+    notes: >-
+      Principal with this permission can list databases. This can be used to discover databases.
```
Comment (nbrahms) [inline services/gcp/datastore/databases.yml:20]: Grammar here too please

File: services/gcp/datastore/databases.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  export:
+    vulnerabilities: [collection:data, impact:spend]
+    notes: >-
+      Principal with this permission can export entities to a Cloud Storage bucket.
```
Comment (nbrahms) [inline services/gcp/datastore/databases.yml:24]: etc.

File: services/gcp/datastore/databases.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    notes: >-
+      Principal with this permission can create a new database. This can consume the quota limits for database creation.
+  update:
+    vulnerabilities: [destruction:defense]
+    notes: >-
+      Principal with this permission can update the database. Updating the database can involve removing delete protection.
+  delete:
+    vulnerabilities: [destruction:data]
+    notes: >-
+      Principal with this permission can delete the database. Deleting the database can involve removing delete protection.
```
Comment (nbrahms) [inline services/gcp/datastore/databases.yml:41]:
```suggestion
      A principal with this permission can delete the database, destroying all data.
```

**Review by nbrahms — APPROVED** [review]
Let's fix the verbiage prior to merge, please.

File: services/gcp/datastore/databases.yml (hunk 5 above)
Comment (varunalla) [reply services/gcp/datastore/databases.yml]: i have verified, it does not share anything sensitive apart from mode of operation , data archival mode.

**Pull request merged**